File: daily/python/Nov-17-container-w-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        res = 0
        while (left < right):
            if height[left] <= height[right]:
                res = max(res, (right-left)*height[left])
                left = left + 1
            else:
                res = max(res, (right-left)*height[right])
                right = right - 1

        return res

File: daily/cpp/Nov-17-container-w-most-water.cpp
#include <vector>
using namespace std;

class Solution {
public:
  int maxArea(vector<int>& height) {
    int res = 0;
    int left = 0, right = height.size() - 1;
    while (left < right) {
      if (left <= right) {
        res = max(res, (right-left) * height[left]);
        left++;
      } else {
        res = max(res, (right-left) * height[right]);
        right--;
      }
    }
    return res;
  }
}
